tests/pyfunc/test_spark.py

```diff
@@ -124,6 +145,61 @@ def test_spark_udf(spark, model_path):
                 assert expected == actual
 
 
+@pytest.mark.parametrize("sklearn_version, expected_result", [('0.22.1', 1.0), ('0.24.0', 2.0)])
+def test_spark_udf_conda_manager_can_restore_env(spark, model_path, sklearn_version, expected_result):
+    class EnvRestoringTestModel(mlflow.pyfunc.PythonModel):
+
+        def __init__(self):
+            pass
+
+        def predict(self, context, model_input):
+            from packaging.version import Version
+            import sklearn
+            if Version(sklearn.__version__) < Version('0.23'):
+                self.is_low_version = True
+            else:
+                self.is_low_version = False
+
+            return model_input.apply(lambda row: (1.0 if self.is_low_version else 2.0), axis=1)
```

Comment:
Can we make this more accurate by returning 1.0 if the version matches the expected version exactly and 0 if it doesn't?